rasa/utils/tensorflow/models.py

```diff
@@ -541,6 +541,22 @@ def batch_to_model_data_format(
                             )
                         )
                         idx += 3
+                    elif is_sparse and number_of_dimensions == 4:
+                        # explicitly substitute last dimension in shape with known
+                        # static value
+                        batch_data[key][sub_key].append(
+                            tf.SparseTensor(
+                                batch[idx],
+                                batch[idx + 1],
+                                [
+                                    batch[idx + 2][0],
+                                    batch[idx + 2][1],
+                                    batch[idx + 2][2],
+                                    feature_dimension,
+                                ],
```

Comment:
isn't it easier: `shape = [batch[idx + 2][i] for i in range(number_of_dimensions - 1)] + [feature_dimension]`? then we don't need the `if` about `number_of_dimensions`